### DISCUSSION TITLE
MercadoPago is not defined

### DISCUSSION THREAD

**Discussion by Pacocc97** [mercadopago/sdk-js#162] [Q&A] [answered]
Hello people, I´m trying to integrate MP cardform to my website. Using Astro js and Im having this issue. 
"MercadoPago is not defined".
I guess my question is. Is this a common error? Am I doing something wrong in my general site config? 
Let me show you my code.
```
<body>
  <style>
    #form-checkout {
      display: flex;
      flex-direction: column;
      max-width: 600px;
    }

    .container {
      height: 18px;
      display: inline-block;
      border: 1px solid rgb(118, 118, 118);
      border-radius: 2px;
      padding: 1px 2px;
    }
  </style>
  <form id="form-checkout">
    <div id="form-checkout__cardNumber" class="container"></div>
    <div id="form-checkout__expirationDate" class="container"></div>
    <div id="form-checkout__securityCode" class="container"></div>
    <input type="text" id="form-checkout__cardholderName" />
    <select id="form-checkout__issuer"></select>
    <select id="form-checkout__installments"></select>
    <input type="email" id="form-checkout__cardholderEmail" />

    <button type="submit" id="form-checkout__submit">Pagar</button>
    <progress value="0" class="progress-bar">Cargando...</progress>
  </form>
  <script src="https://sdk.mercadopago.com/js/v2"></script>
  <script>
    const mp = new MercadoPago("PUBLIC_KEY 'Im using my real key in my site'");
  </script>
</body>

```

**Comment by icaldana** [comment, marked as answer]
Hello @Pacocc97,
First of all we apologize for the delay. Secondly, we have tested this script using a test PUBLIC_KEY and it is working normally. Perhaps it is a conflict with AstroJS? Are you still facing this issue?

**We have an official channel that is super prepared to solve your case quickly.**
So, if the error still persists, gather all the information and go to the following link to share the issue with us:
https://www.mercadopago.com/developers/pt/support/center/contact
Thank you very much for your contact!

**Reply by Pacocc97** [reply]
Is working now!! thanks for the response. You were right, it was an astro js issue

**Reply by nerita87** [reply]
Hello @icaldana 
I am using the sdk v2 in a .js file and I have the same error. I import the library in the body of the html as indicated in the documentation and it shows me the error: "Uncaught ReferenceError: MercadoPago is not defined".
Please if you could help me!

**Reply by Pacocc97** [reply]
@nerita87 Are you using a framework?

**Reply by nerita87** [reply]
@Pacocc97 it already works fine. It was cache in my symfony proyect. Thank you!
